automation_tools/satellite6/upgrade/capsule.py

```diff
@@ -97,9 +90,8 @@ def satellite6_capsule_upgrade(cap_host):
     # org, and content host in one), the content host will be unregistered as
     # part of the upgrade process.
     if to_version == '6.2':
-        ak_name = os.environ.get('CAPSULE_SUBSCRIPTION').split(',')[2].strip(
-            ) if os.environ.get('CAPSULE_SUBSCRIPTION') else os.environ.get(
-            'RHEV_CAPSULE_AK')
+        ak_name = os.environ.get('CAPSULE_AK') if os.environ.get(
+            'CAPSULE_AK') else os.environ.get('RHEV_CAPSULE_AK')
```

Comment:
in the first case you cant because you operating on it. But on second you could:

```python
Python 3.5.2 (default, Oct 14 2016, 12:54:53) 
[GCC 6.2.1 20160916 (Red Hat 6.2.1-2)] on linux
In[2]: import os
In[3]: os.environ
Out[3]: 
environ({'SHLVL': '2', 'HOME': '/home/renzo', 'XDG_SESSION_TYPE': 'x11', 'GNOME_DESKTOP_SESSION_ID': 'this-is-deprecated', 'HISTSIZE': '1000', 'DBUS_SESSION_BUS_ADDRESS': 'unix:path=/run/user/1000/bus', 'XDG_MENU_PREFIX': 'gnome-', 'PWD': '/home/renzo/PycharmProjects/sanbox35', 'LOGNAME': 'renzo', 'GDMSESSION': 'gnome', 'WINDOWPATH': '2', 'QTDIR': '/usr/lib64/qt-3.3', 'HOSTNAME': 'note', 'QTLIB': '/usr/lib64/qt-3.3/lib', 'GJS_DEBUG_TOPICS': 'JS ERROR;JS LOG', 'LESSOPEN': '||/usr/bin/lesspipe.sh %s', 'MODULEPATH': '/etc/scl/modulefiles:/usr/share/Modules/modulefiles:/etc/modulefiles:/usr/share/modulefiles', 'XDG_VTNR': '2', 'PYTHONDONTWRITEBYTECODE': '1', 'QT_QPA_PLATFORMTHEME': 'qgnomeplatform', 'QT_IM_MODULE': 'ibus', 'XDG_CURRENT_DESKTOP': 'GNOME', 'SESSION_MANAGER': 'local/unix:@/tmp/.ICE-unix/2194,unix/unix:/tmp/.ICE-unix/2194', 'GIO_LAUNCHED_DESKTOP_FILE_PID': '3195', 'XAUTHORITY': '/run/user/1000/gdm/Xauthority', 'XDG_SEAT': 'seat0', 'PATH': '/usr/lib64/qt-3.3/bin:/usr/lib64/ccache:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/renzo/.local/bin:/home/renzo/bin', 'BASH_FUNC_module()': '() {  eval `/usr/bin/modulecmd bash $*`\n}', 'IPYTHONENABLE': 'True', 'GJS_DEBUG_OUTPUT': 'stderr', 'XDG_SESSION_DESKTOP': 'gnome', 'LANG': 'en_US.UTF-8', 'GDM_LANG': 'en_US.UTF-8', '_': '/home/renzo/bin/pycharm-2016.3/jre/jre/bin/java', 'USER': 'renzo', 'PYTHONIOENCODING': 'UTF-8', 'BASH_FUNC_scl()': '() {  local CMD=$1;\n if [ "$CMD" = "load" -o "$CMD" = "unload" ]; then\n eval "module $@";\n else\n /usr/bin/scl "$@";\n fi\n}', 'NLSPATH': '/usr/dt/lib/nls/msg/%L/%N.cat', 'MODULESHOME': '/usr/share/Modules', 'QTINC': '/usr/lib64/qt-3.3/include', 'XDG_SESSION_ID': '2', 'XFILESEARCHPATH': '/usr/dt/app-defaults/%L/Dt', 'XDG_RUNTIME_DIR': '/run/user/1000', 'LD_LIBRARY_PATH': '/home/renzo/bin/pycharm-2016.3/bin:', 'XMODIFIERS': '@im=ibus', 'PYCHARM_HOSTED': '1', 'SHELL': '/bin/bash', 'PYTHONPATH': '/home/renzo/bin/pycharm-2016.3/helpers/pydev', 'DISPLAY': ':0', 'PYTHONUNBUFFERED': '1', 'SSH_AUTH_SOCK': '/run/user/1000/keyring/ssh', 'USERNAME': 'renzo', 'LOADEDMODULES': '', 'GIO_LAUNCHED_DESKTOP_FILE': '/home/renzo/.local/share/applications/jetbrains-pycharm.desktop', 'DESKTOP_SESSION': 'gnome', 'MAIL': '/var/spool/mail/renzo', 'HISTCONTROL': 'ignoredups'})
In[4]: os.environ['RHEV_CAPSULE_AK']
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2881, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-104a9375923e>", line 1, in <module>
    os.environ['RHEV_CAPSULE_AK']
  File "/usr/lib64/python3.5/os.py", line 725, in __getitem__
    raise KeyError(key) from None
KeyError: 'RHEV_CAPSULE_AK'

In[5]: os.environ['RHEV_CAPSULE_AK']= 'Some Fall Back Value'
In[6]: os.environ['RHEV_CAPSULE_AK']
Out[6]: 
'Some Fall Back Value'
In[7]: fallback={'CAPSULE_AK': os.environ['RHEV_CAPSULE_AK']}
In[8]: fallback
Out[8]: 
{'CAPSULE_AK': 'Some Fall Back Value'}

In[10]: from collections import ChainMap
   ...: env_with_fallback=ChainMap(os.environ, fallback)

In[12]: env_with_fallback['HOME']  # it is on environment
Out[12]: 
'/home/renzo'
In[13]: env_with_fallback['CAPSULE_AK']  # it is on fallback
Out[13]: 
'Some Fall Back Value'
In[14]: os.environ['CAPSULE_AK']='another value'
In[15]: env_with_fallback['CAPSULE_AK']  # now it is on environment
Out[15]: 
'another value'
In[16]: env_with_fallback['foo']  # not on environ nor fallback
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2881, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-16-4bac41b8d7f7>", line 1, in <module>
    env_with_fallback['foo']  # not on environ nor fallback
  File "/usr/lib64/python3.5/collections/__init__.py", line 878, in __getitem__
    return self.__missing__(key)            # support subclasses that define __missing__
  File "/usr/lib64/python3.5/collections/__init__.py", line 870, in __missing__
    raise KeyError(key)
KeyError: 'foo'
```

In fact all the logic of extraction of env var could be refactored in one place. But maybe in another time ;)